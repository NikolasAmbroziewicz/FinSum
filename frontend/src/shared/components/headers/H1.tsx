import { IHeaderProp, Variant, Position } from './Header.types';
import { useHeader } from './useHeader';

const H1: React.FC<IHeaderProp> = ({
  children,
  position = Position.center,
  variant = Variant.dark,
  styles = ''
}) => {
  const {textColor, textPosition} = useHeader(variant, position)

  return (
    <h1 className={`${textColor()} ${textPosition()} text-3xl ${styles}`}>
      {children}
    </h1>
  );
};

export default H1;
