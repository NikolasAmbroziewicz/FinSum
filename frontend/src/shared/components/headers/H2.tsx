import { IHeaderProp, Variant, Position } from './Header.types';

const H2: React.FC<IHeaderProp> = ({
  children,
  variant = Variant.dark,
  position = Position.center,
  styles= ''
}) => {
  const textColor = () => {
    return variant === Variant.dark ? 'text-gray-600' : 'text-stone-50';
  };

  const textPosition = () => {
    switch(position) {
      case Position.center:
        return 'text-center'
      case Position.left:
        return 'text-left'
      case Position.right:
        return 'text-right'
    }
  }

  return <h2 className={`${textColor()} ${textPosition()} text-xl, ${styles}`}>{children}</h2>;
};

export default H2;
