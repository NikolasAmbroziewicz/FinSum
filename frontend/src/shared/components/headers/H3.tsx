import { IHeaderProp, Variant, Position } from './Header.types';

const H3: React.FC<IHeaderProp> = ({
  children,
  variant = Variant.dark,
  position = Position.center,
  styles=''
}) => {
  const textColor = () => {
    return variant === Variant.dark ? 'text-gray-600' : 'text-stone-50';
  };

  const textPosition = () => {
    switch(position) {
      case Position.center:
        return 'text-center'
      case Position.left:
        return 'text-left'
      case Position.right:
        return 'text-right'
    }
  }

  return <h3 className={`${textColor()} ${textPosition()} text-base ${styles}`}>{children}</h3>;
};

export default H3;
