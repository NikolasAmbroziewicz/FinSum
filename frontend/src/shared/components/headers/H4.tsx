import { IHeaderProp, VariantColor } from './Header.types';

const H4: React.FC<IHeaderProp> = ({
  children,
  variant = VariantColor.dark
}) => {
  const textColor = () => {
    return variant === VariantColor.dark ? 'text-gray-600' : 'text-stone-50';
  };
  return <h4 className={`text-center ${textColor()} text-sm`}>{children}</h4>;
};

export default H4;
