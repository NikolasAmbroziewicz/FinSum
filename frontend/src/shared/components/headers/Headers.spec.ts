import { describe, expect, it } from 'vitest';
import H1 from './H1';
import H2 from './H2';
import H3 from './H3';
import H4 from './H4';
import { mockHeadersBaseProps } from './Header.mocks';

describe('Headers', () => {
  it('H1 define', () => {
    expect(H1(mockHeadersBaseProps.base)).toMatchSnapshot();
  });

  it('H2 define', () => {
    expect(H2(mockHeadersBaseProps.base)).toMatchSnapshot();
  });

  it('H3 define', () => {
    expect(H3(mockHeadersBaseProps.base)).toMatchSnapshot();
  });

  it('H4 define', () => {
    expect(H4(mockHeadersBaseProps.base)).toMatchSnapshot();
  });
});
