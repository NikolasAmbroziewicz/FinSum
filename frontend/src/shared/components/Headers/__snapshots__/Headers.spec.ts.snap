// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Headers > H1 define 1`] = `
<h1
  className="text-gray-600 text-center  text-3xl "
>
  This is awesome Header Text
</h1>
`;

exports[`Headers > H2 define 1`] = `
<h2
  className="text-gray-600 text-center text-2xl "
>
  This is awesome Header Text
</h2>
`;

exports[`Headers > H3 define 1`] = `
<h3
  className="text-gray-600 text-center text-xl "
>
  This is awesome Header Text
</h3>
`;

exports[`Headers > H4 define 1`] = `
<h4
  className="text-gray-600 text-center text-base "
>
  This is awesome Header Text
</h4>
`;
